kenvandine/xournalpp-snap
# make LaTeX work via TinyTeX

# Pull Request

## Description

This PR makes the LaTeX tool in Xournal++ work. It replaces the installation of `texlive-extra` by TinyTeX, which is a portable version, a subset of the `TeXLive` distribution, that is aimed at using less space.

Fixes #3

## Type of change


## How Has This Been Tested?

**Test Configuration**:

- Build the PR via `snapcraft` and install it
- remove the file `~/snap/xournalpp/current/.config/xournalpp/settings.xml` or modify the line
 ``` 
<property name="latexSettings.globalTemplatePath" value="/snap/xournalpp/x10/usr/share/xournalpp/resources/default_template.tex"/>
```  
in it to refer to the current version (replace `x10` by `current` in the example)
Alternatively update the global LaTeX template path via the GUI in Xournal++ (menu `Edit > Preferences`, LaTeX panel)

- `snap run xournalpp`
- Press the LaTeX Tool button 
![grafik](https://user-images.githubusercontent.com/40485433/128021331-5929e540-58a1-4d53-8616-99517a58e186.png)
and observe that the tool works and the formula can be modified
![grafik](https://user-images.githubusercontent.com/40485433/128019918-72613c02-d1b7-43d5-aa42-1f2a6738f613.png)

* OS: Ubuntu 21.04

**Additional information**:
- It is easy to add more LaTeX packages in the `snapcraft.yaml` if required (via the `tlmgr install` command)
- It would be possible to use the `TinyTeX-0` (bare minimum) release instead of `TinyTeX-1` and enrich it with hand-picked packages
- It would also be possible to use the bigger `TinyTeX` release (and enrich it with some hand-picked packages)

## Checklist:
